### DISCUSSION TITLE
Can I access the express handler of acebase-server and/or pass a custom express server to it?

### DISCUSSION THREAD

**Discussion by trenta3** [appy-one/acebase-server#16] [Q&A] [answered]
I'm sorry for the amount of questions that I am asking in these days, but I am evaluating the usage of acebase for a project of mine and I would like to understand better what are the tradeoffs involved and the easyness of doing various operations.

I would like to use SvelteKit to generate the webapp, and serve acebase on the same server serving webpages (it is for a small project and I would like it to be as self-contained as it can get). To do this using [SK hooks](https://kit.svelte.dev/docs/hooks#handle) requires to have an handler from acebase-server to call, but as I can see in [acebase-server sources](https://github.com/appy-one/acebase-server/blob/36cfec41ca6822d84fbe51f61e20f2b0ffdcbef0/src/server.ts#L139), the express application is not stores as an accessible property of the server class.

Would it be possible for it to be made visible?

Edit: I am aware that I can extend the server application with custom routes, but those end up under `/ext/` and thus it may not be possible to serve pages the way SvelteKit does.

**Comment by appy-one** [comment]
Sure, I'll take a look at this tomorrow

**Comment by appy-one** [comment, marked as answer]
Done, the Express app is now exposed through the `app` property on the server instance. See [this commit](https://github.com/appy-one/acebase-server/commit/38c704a22cf080fae610354b89b9e2cfc0e350c1)

Example usage:
```js
server.app.get('/hello', (req, res) => {
    res.send('Hi!');
});
```

**Reply by trenta3** [reply]
Thanks! I wil try it in the next days

**Reply by trenta3** [reply]
It works very well. It would be even more cool if one could create the AceBaseServer separately from starting it, so that I could expose multiple AceBase databases on the same express app (if I understood correctly the routes depend on the dbname).
I see in `acebase-server/src/server.ts` that the constructor method directly calls the `init` method, which sets up middlewares **and** also starts the server. Separating the latter in its own `start` method would allow such scenarios.
